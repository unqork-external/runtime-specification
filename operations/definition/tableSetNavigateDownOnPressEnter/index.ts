export * from './tableSetNavigateDownOnPressEnter.model'
export * from './tableSetNavigateDownOnPressEnter.options'
