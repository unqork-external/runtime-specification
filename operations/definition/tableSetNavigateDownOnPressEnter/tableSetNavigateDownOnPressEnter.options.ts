import { Description, Required } from '@tsed/schema'

import { TrimmedDescription } from '../../../../decorators/schema/trimmedDescription.decorator'

export class TableSetNavigateDownOnPressEnterOperationOptions {
  @Required()
  @Description('The key of the Table component to set navigation options.')
  targetKey: string

  @Required()
  @TrimmedDescription(`
    When enabled, pressing Enter on a cell in edit mode sets the focus to the cell immediately below.
    The new selected cell changes to edit mode.
    If there's no cell below the current cell, nothing happens.
  `)
  enabled: boolean
}
