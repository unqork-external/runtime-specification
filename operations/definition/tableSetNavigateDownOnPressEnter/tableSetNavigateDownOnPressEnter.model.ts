import { DiscriminatorValue, Required } from '@tsed/schema'

import { TableSetNavigateDownOnPressEnterOperationOptions } from './tableSetNavigateDownOnPressEnter.options'
import { OperationTypes } from '../../enums/operation-types.enum'
import { Operation } from '../../interface/operations.interface'

@DiscriminatorValue(OperationTypes.TABLE_SET_NAVIGATE_DOWN_ON_PRESS_ENTER)
// eslint-disable-next-line max-len
export class TableSetNavigateDownOnPressEnterOperation extends Operation<OperationTypes.TABLE_SET_NAVIGATE_DOWN_ON_PRESS_ENTER> {
  public type = OperationTypes.TABLE_SET_NAVIGATE_DOWN_ON_PRESS_ENTER as const

  @Required()
  public options: TableSetNavigateDownOnPressEnterOperationOptions =
    new TableSetNavigateDownOnPressEnterOperationOptions()
}
