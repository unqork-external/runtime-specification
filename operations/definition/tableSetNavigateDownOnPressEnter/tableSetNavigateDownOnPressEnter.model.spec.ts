import { TableSetNavigateDownOnPressEnterOperation } from './tableSetNavigateDownOnPressEnter.model'
import { generateSchemaAndValidate } from '../../../../utilities'

describe('Operation Model: TableSetNavigateDownOnPressEnter Tests', () => {
  let validate, schema
  beforeAll(() => {
    const response = generateSchemaAndValidate(TableSetNavigateDownOnPressEnterOperation)
    validate = response.validate
    schema = response.schema
  })

  it('should match this JSON Schema', () => {
    expect(schema).toMatchSnapshot()
  })
})
