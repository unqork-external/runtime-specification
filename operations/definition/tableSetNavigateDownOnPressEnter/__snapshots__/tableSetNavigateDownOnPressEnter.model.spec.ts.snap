// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Operation Model: TableSetNavigateDownOnPressEnter Tests should match this JSON Schema 1`] = `
{
  "definitions": {
    "TableSetNavigateDownOnPressEnterOperationOptions": {
      "properties": {
        "enabled": {
          "description": "When enabled, pressing Enter on a cell in edit mode sets the focus to the cell immediately below. The new selected cell changes to edit mode. If there's no cell below the current cell, nothing happens.",
          "type": "boolean",
        },
        "targetKey": {
          "description": "The key of the Table component to set navigation options.",
          "minLength": 1,
          "type": "string",
        },
      },
      "required": [
        "targetKey",
        "enabled",
      ],
      "type": "object",
    },
  },
  "description": "When the Table component is in edit mode, this will change the navigation behavior to traverse down cells when the end-user presses 'Enter'.",
  "properties": {
    "creatorSummary": {
      "description": "A detailed summary of the operation",
      "type": "string",
    },
    "name": {
      "description": "Name of the operation",
      "type": "string",
    },
    "options": {
      "$ref": "#/definitions/TableSetNavigateDownOnPressEnterOperationOptions",
    },
    "type": {
      "const": "TABLE_SET_NAVIGATE_DOWN_ON_PRESS_ENTER",
      "examples": [
        "TABLE_SET_NAVIGATE_DOWN_ON_PRESS_ENTER",
      ],
      "type": "string",
    },
  },
  "required": [
    "options",
  ],
  "stability": "STABLE",
  "type": "object",
}
`;
